retry/config.go

```diff
@@ -68,13 +68,17 @@ func (c Configuration) NewRetrier(scope tally.Scope) Retrier {
 	if c.MaxRetries != 0 {
 		maxRetries = c.MaxRetries
 	}
-	retryOpts := NewOptions().
+	return NewOptions().
```

Comment:
btw I know this is not your change, but `c.Forever` and `c.Jitter` should probably be *bool in the config, otherwise you can't use the default and always need to pick one